internal/channels/notify.go

```diff
@@ -24,12 +30,15 @@ func NewNotifier(ctx context.Context) *Notifier {
 // Subscribe to the notifier.
 //
 // The returned channel will be closed when the notifier context is cancelled.
-func (b *Notifier) Subscribe() <-chan struct{} {
+func (b *Notifier) Subscribe(ctx context.Context) <-chan struct{} {
```

Comment:
A bit late to the party, but how is this different to pubsub.Topic?

Reply:
I did initially use `pubsub.Topic` with raft, but there was something that made it unsuitable. Should have really written it in the comments...

I can see what was the problem by using `pubsub.Topic` as the implementation and checking what fails

Reply:
I think the difference here is that notifier drops messages if the subscribers are full, while pubsub would block